grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALExploreSchemaConcept.java

```diff
@@ -41,29 +42,31 @@
  * Class used to build the HAL representation of a given concept.
  */
 
-class HALExploreType extends HALExploreConcept{
+class HALExploreSchemaConcept extends HALExploreConcept {
 
-    HALExploreType(Concept concept, Keyspace keyspace, int offset, int limit) {
+    HALExploreSchemaConcept(Concept concept, Keyspace keyspace, int offset, int limit) {
         super(concept, keyspace, offset, limit);
     }
 
     void populateEmbedded(Representation halResource, Concept concept) {
 
-        Type type = concept.asType();
+        SchemaConcept schemaConcept = concept.asSchemaConcept();
 
-        // Role types played by current type
-        attachRolesPlayed(halResource, type.plays());
-        // Resources types owned by the current type
-        attachTypeResources(halResource, type);
-        // Subtypes
-        attachSubTypes(halResource, type);
+        if (schemaConcept.isType()) {
+            // Role types played by current type
+            attachRolesPlayed(halResource, schemaConcept.asType().plays());
+            // Resources types owned by the current type
+            attachTypeResources(halResource, schemaConcept.asType());
+            // Subtypes
+            attachSubTypes(halResource, schemaConcept.asType());
```

Comment:
A schema concept does still have sub-concepts, so this can go outside of the `if` (unless `roleTypeSchema` handles this below)